samples/ClassroomLabs/Module/Library/Az.AzureLabs.psm1

```diff
@@ -0,0 +1,1282 @@
+
+                if ($null -ne $variable)
+                {
+                    if ($SessionState -eq $ExecutionContext.SessionState)
+                    {
+                        Set-Variable -Scope 1 -Name $variable.Name -Value $variable.Value -Force -Confirm:$false -WhatIf:$false
+                    }
+                    else
+                    {
+                        $SessionState.PSVariable.Set($variable.Name, $variable.Value)
+                    }
+                }
+            }
+        }
+
+        if ($PSCmdlet.ParameterSetName -eq 'Filtered')
+        {
+            foreach ($varName in $filterHash.Keys)
+            {
+                if (-not $vars.ContainsKey($varName))
+                {
+                    $variable = $Cmdlet.SessionState.PSVariable.Get($varName)
+
+                    if ($null -ne $variable)
+                    {
+                        if ($SessionState -eq $ExecutionContext.SessionState)
+                        {
+                            Set-Variable -Scope 1 -Name $variable.Name -Value $variable.Value -Force -Confirm:$false -WhatIf:$false
+                        }
+                        else
+                        {
+                            $SessionState.PSVariable.Set($variable.Name, $variable.Value)
+                        }
+                    }
+                }
+            }
+        }
+
+    } # end
+
+} # function Get-CallerPreference
+
+function PrintHashtable {
+  param($hash)
+
+  return ($hash.Keys | ForEach-Object { "$_ $($hash[$_])" }) -join "|"
+}
+
+# Taken from https://gallery.technet.microsoft.com/scriptcenter/ConvertFrom-ISO8601Duration-704763e0
+function ConvertFrom-ISO8601Duration {
+    
+  [CmdletBinding(SupportsShouldProcess=$false)]
+  [OutputType([System.TimeSpan])]
+
+  param(
+      [Parameter(ValueFromPipeline=$true, ValueFromPipelineByPropertyName=$true, Mandatory=$true)]
+      [Alias('ISO8601', 'String')]
+      [string]$Duration
+  )
+
+  $pattern = '^P?T?((?<Years>\d+)Y)?((?<Months>\d+)M)?((?<Weeks>\d+)W)?((?<Days>\d+)D)?(T((?<Hours>\d+)H)?((?<Minutes>\d+)M)?((?<Seconds>\d*(\.)?\d*)S)?)$'
+
+  if($Duration -match $pattern) {
+      Set-StrictMode -Off
+      $dt = [datetime]::MinValue
+      Write-Verbose (PrintHashtable -hash $Matches)
+
+      if ($Matches.Seconds) { $dt = $dt.AddSeconds($Matches.Seconds) }
+      if ($Matches.Minutes) { $dt = $dt.AddMinutes($Matches.Minutes) }
+      if ($Matches.Hours)   { $dt = $dt.AddHours($Matches.Hours) }
+      if ($Matches.Days)    { $dt = $dt.AddDays($Matches.Days) }
+      if ($Matches.Weeks)   { $dt = $dt.AddDays(7*$Matches.Weeks) }
+      if ($Matches.Months)  { $dt = $dt.AddMonths($Matches.Months) }
+      if ($Matches.Years)   { $dt = $dt.AddYears($Matches.Years) }
+      $dt - [datetime]::MinValue
+  } else {
+      Write-Warning 'The provided string does not match the ISO 8601 duration format'
+  }
+}
+
+function Get-AzureRmCachedAccessToken()
+{
+  $ErrorActionPreference = 'Stop'
```

Comment:
nit: Can you make the indentations in this file consistent? Some parts at 2 spaces, other 4 spaces.

Reply:
Done shift-alt-f in vscode to do automatic formatting.